platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -318,24 +323,85 @@ private List<EnvVar> getEnvironmentVariables(
         StaticConfig.VALUE_SERIALIZER,
         getEnvVariableFromNode(streamOut.getSpec(), StaticConfig.VALUE_SERIALIZER));
 
-    List<EnvVar> variables = new ArrayList<>();
-    variables.add(createEnvVariable(StaticConfig.STREAM_IN_CONFIG_NAME, streamIn.getName()));
-    variables.add(createEnvVariable(StaticConfig.STREAM_OUT_CONFIG_NAME, streamOut.getName()));
+    // Store Serde-related information
+    Map<String, Object> dataCaterConfig = new HashMap<>();
+    dataCaterConfig.put(
+        StaticConfig.KEY_DESERIALIZER, streamInConfig.get(StaticConfig.KEY_DESERIALIZER));
+    dataCaterConfig.put(
+        StaticConfig.VALUE_DESERIALIZER, streamInConfig.get(StaticConfig.VALUE_DESERIALIZER));
+    dataCaterConfig.put(
+        StaticConfig.KEY_SERIALIZER, streamOutConfig.get(StaticConfig.KEY_SERIALIZER));
+    dataCaterConfig.put(
+        StaticConfig.VALUE_SERIALIZER, streamOutConfig.get(StaticConfig.VALUE_SERIALIZER));
```

Comment:
Is there a reason we do not use `Map.of(K1, V1, K2, V2, ...)` for initiating the `Map` `datacaterConfig`?

Reply:
Nope. Thanks for the hint! Fixed with 985e148bdc9efbf4745d603baa2faf1d9b49254a